teps/0114-custom-tasks-beta.md

```diff
@@ -208,6 +209,19 @@ Remove guarding of `Custom Tasks` behind `enable-custom-tasks` and `enable-api-f
 When [TEP-0096: Pipelines V1 API][tep-0096] is implemented to add V1 API, `Custom Tasks` will be gated behind feature
 gate `enable-api-fields` being set to `"beta"` - this is out of scope for this TEP  (in scope for TEP-0096).
 
+##### Cancellation
+
+Implementing cancellation is now `required` instead of `optional`. If the Custom Task implementor does not support cancellation via `.spec.status`, `Pipeline` timeouts and `PipelineRun` cancellation will not work properly.
+
+Pipeline Controller sets the `spec.Status` and `spec.StatusMessage` to signal `CustomRuns` about the `Cancellation`, while `CustomRun` controller **MUST** update its `status.conditions` as following once noticed the change on `spec.Status`.
```

Comment:
When does pipeline controller set the status about the cancellation? Is it only when it times out or does it also include all types of cancellation that a pipelineRun supports?

https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#cancelling-a-pipelinerun
https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#gracefully-cancelling-a-pipelinerun
https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#gracefully-stopping-a-pipelinerun

Reply:
Thanks for asking Priti! It includes `Timeout` and `Cancellation` ([cancelling-a-pipelinerun](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#cancelling-a-pipelinerun) and [gracefully-cancelling-a-pipelinerun](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#gracefully-cancelling-a-pipelinerun)[^1], notice [gracefully-stopping-a-pipelinerun](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#gracefully-stopping-a-pipelinerun) is not included.)
 
 [^1]: [`cancelPipelineTaskRuns`](https://github.com/tektoncd/pipeline/blob/b15a80e5ab9299386024578ac25b1f42c4d364ac/pkg/reconciler/pipelinerun/cancel.go#L126) is only used in the two kinds of cancellation: [Cancelling a PipelineRun](https://github.com/tektoncd/pipeline/blob/b15a80e5ab9299386024578ac25b1f42c4d364ac/pkg/reconciler/pipelinerun/cancel.go#L9), [Gracefully cancelling a PipelineRun](https://github.com/tektoncd/pipeline/blob/b15a80e5ab9299386024578ac25b1f42c4d364ac/pkg/reconciler/pipelinerun/cancel.go#L205)

Reply:
Updated the doc. LMK if there's anyother concern!

Reply:
Thank you @XinruZhang 👍